src/logic/voice/voice.ts

```diff
@@ -73,10 +78,88 @@ export function createVoiceComponent(
     await voiceDB.updateUserStatusAsConnectionInterrupted(userAddress, roomName)
   }
 
+  /**
+   * Handles the event when a participant joins a COMMUNITY voice chat room.
+   */
+  async function handleCommunityParticipantJoined(userAddress: string, roomName: string): Promise<void> {
+    logger.debug(`Community participant joined: ${userAddress} in room ${roomName}`)
+    // Simply update the user status to connected
+    await voiceDB.updateCommunityUserStatus(userAddress, roomName, VoiceChatUserStatus.Connected)
+  }
+
+  /**
+   * Handles the event when a participant leaves a COMMUNITY voice chat room.
+   */
+  async function handleCommunityParticipantLeft(
+    userAddress: string,
+    roomName: string,
+    disconnectReason: DisconnectReason
+  ): Promise<void> {
+    logger.debug(`Community participant left: ${userAddress} in room ${roomName}, reason: ${disconnectReason}`)
+
+    if (disconnectReason === DisconnectReason.DUPLICATE_IDENTITY) {
+      logger.debug('Ignoring disconnect due to duplicate identity')
+      return
+    }
+
+    if (disconnectReason === DisconnectReason.ROOM_DELETED) {
+      await voiceDB.deleteCommunityVoiceChat(roomName)
+      return
+    }
+
+    // Update user status based on disconnect reason
+    if (disconnectReason === DisconnectReason.CLIENT_INITIATED) {
+      await voiceDB.updateCommunityUserStatus(userAddress, roomName, VoiceChatUserStatus.Disconnected)
+    } else {
+      await voiceDB.updateCommunityUserStatus(userAddress, roomName, VoiceChatUserStatus.ConnectionInterrupted)
+    }
+
+    // Check if community room should be destroyed (no moderators for 5 mins)
+    // NOTE: This only triggers when someone disconnects. The main cleanup is handled by expireCommunityVoiceChats job
+    if (await voiceDB.shouldDestroyCommunityRoom(roomName)) {
+      logger.debug(`Community room ${roomName} should be destroyed, deleting from livekit`)
+      await livekit.deleteRoom(roomName)
+      await voiceDB.deleteCommunityVoiceChat(roomName)
+
+      const communityId = getCommunityIdFromRoomName(roomName)
+      analytics.fireEvent(AnalyticsEvent.EXPIRE_CALL, {
+        call_id: communityId
+      })
+    }
```

Comment:
As this is checking that what the job does when a participant disconnects, we might be able to remove this section, specially taking into consideration that there's a grace period being checked in the `shouldDestroyCommunityRoom` function. What I think we're missing is to destroy the community room when there are no other active moderators an the participant leaving the room voluntarily is a moderator, what do you think if we change what we have here for that?